src/MicroPlate.php

```diff
@@ -27,18 +34,133 @@ class MicroPlate
+    {
+        $coordinate = $this->nextFreeWellCoordinate($flowDirection);
+        $this->addWell($coordinate, $content);
+    }
+
+    public function clearWells(): void
+    {
+        $this->wells = new Collection([]);
+        foreach ($this->coordinateSystem->columns() as $column) {
+            foreach ($this->coordinateSystem->rows() as $row) {
+                $this->wells->put($row . $column, self::EMPTY_WELL);
+            }
+        }
+    }
+
+    /**
+     * @param Coordinate<TCoordinateSystem> $coordinate
+     */
+    public function wellIsEmpty(Coordinate $coordinate): bool
+    {
+        return self::EMPTY_WELL === $this->wells[$coordinate->toString()];
+    }
+
+    /**
+     * If well is empty it will add the content.
+     * If well is not empty, it will overwrite the old content with the new content.
+     *
+     * @param Coordinate<TCoordinateSystem> $coordinate
+     * @param TWell $content
+     */
+    public function forceAddWell(Coordinate $coordinate, $content): void
+    {
+        $this->wells->put($coordinate->toString(), $content);
+    }
+
+    /**
+     * @return WellsCollection
+     */
+    public function sortedWells(FlowDirection $flowDirection): Collection
+    {
+        return $this->wells->sortBy(function ($value, $key) use ($flowDirection) {
+            switch ($flowDirection->getValue()) {
+                case FlowDirection::ROW:
+                    return $key;
+                case FlowDirection::COLUMN:
+                    $row = Str::of($key)->substr(0, 1);
+                    $column = Str::of($key)->substr(1, 2);
+
+                    return $column . $row;
+                // @codeCoverageIgnoreStart all Enums are listed and this should never happen
+                default:
+                    throw new UnexpectedFlowDirection($flowDirection);
+                // @codeCoverageIgnoreEnd
+            }
+        }, SORT_NATURAL);
+    }
+
+    /**
+     * @return WellsCollection
+     */
+    public function freeWells(): Collection
+    {
+        return $this->wells->filter(fn ($value, $key) => self::EMPTY_WELL === $value);
+    }
+
+    /**
+     * @return WellsCollection
+     */
+    public function filledWells(): Collection
+    {
+        return $this->wells->filter(fn ($value, $key) => self::EMPTY_WELL !== $value);
+    }
+
+    /**
+     * @return Coordinate<TCoordinateSystem>
+     */
+    public function nextFreeWellCoordinate(FlowDirection $flowDirection): Coordinate
+    {
+        $coordinateString = $this->sortedWells($flowDirection)
+            ->filter(fn ($value, $key) => self::EMPTY_WELL === $value)
+            ->keys()
+            ->first();
+
+        if (! is_string($coordinateString)) {
+            throw new MicroPlateFullException('No free spots left on plate');
+        }
+
+        /** @var Coordinate<TCoordinateSystem> $coordinate */
+        $coordinate = Coordinate::fromString(
```

Comment:
Method needs to have a generic second argument, then the type assertion above becomes unnecessary

Reply:
The template param had to be in the static method, fixed it for you.